src/python/pants/backend/python/typecheck/pyright/rules.py

```diff
@@ -110,17 +185,22 @@ async def pyright_typecheck_partition(
         ),
     )
 
-    # venv workaround as per: https://github.com/microsoft/pyright/issues/4051
-    dummy_config_digest = await Get(
-        Digest,
-        CreateDigest(
-            [
-                FileContent(
-                    "pyrightconfig.json",
-                    f'{{ "venv": "{requirements_venv_pex.venv_rel_dir}" }}'.encode(),
-                )
-            ]
-        ),
+    source_roots_get = Get(
+        SourceRootsResult,
+        SourceRootsRequest,
+        SourceRootsRequest.for_files(root_sources.snapshot.files),
+    )
+
+    requirements_venv_pex, source_roots = await MultiGet(
+        requirements_venv_pex_get,
+        source_roots_get,
+    )
+
+    # Patch the config file to use the venv directory from the requirements pex,
+    # and add source roots to the `extraPaths` key in the config file.
+    unique_source_roots = {root.path for root in source_roots.path_to_root.values()}
+    patched_config_digest = await _patch_config_file(
```

Comment:
How do the user-added args play alongside the configuration? What should happen? Feels like an escape hatch situation.